frontend/src/views/producer/dashboard/edition-form/Editor.jsx

```diff
@@ -49,82 +60,137 @@ const RedButton = styled(Button)`
   }
 `;
 
-const defaulTicketTypeValue = { ...TicketCatalog.create().toJSON() };
-
-const Editor = observer(
-  class Editor extends React.Component {
-    eventDraft = observable({
-      ...EventModel.create({ ticketCatalog: [defaulTicketTypeValue] }).toJSON(),
-    });
-
-    componentDidMount() {
-      const { palette } = this.props.theme;
-      // override specific class name from ant-design
-      // THIS IS A HACK
-      injectGlobal`
+class Editor extends React.Component {
+  internalData = observable({
+    creationMode: true,
+  });
+  constructor(props) {
+    super(props);
+    this.internalData.eventDraft = props.selectedEvent
```

Comment:
Just for curiosity... What is the reason for using internalData here instead of the state of the Component?

Reply:
react will complain that i am directly modifying state instead of using setState. The component uses mobx for reactivity, not internal state, so im using `internalData` instead

Reply:
quite arbitrary naming though, im up for a rename. Can you suggest one?

Reply:
how about calling it editorState?

Reply:
btw, Shouldn't you use `@observer` and `@observable` decorators as done here: https://alexhisen.gitbooks.io/mobx-recipes/content/use-observables-instead-of-state-in-react-components.html

Reply:
I used the exact samething, just not using decorator

Reply:
ok 👍